vsintegration/src/vs/FsPkgs/FSharp.Project/Common.Source.CSharp/Project/HierarchyNode.cs

```diff
@@ -3301,72 +3300,61 @@ string newTargetFrameworkMoniker
             projectMgr.BuildProject.SetProperty(ProjectFileConstants.TargetFrameworkVersion, HierarchyNode.GetFrameworkVersionString(frameworkName));
             projectMgr.BuildProject.SetProperty(ProjectFileConstants.TargetFrameworkProfile, frameworkName.Profile);
 
+            string targetFSharpCoreVersion = projectMgr.BuildProject.GetPropertyValue(ProjectFileConstants.TargetFSharpCoreVersion);
+            bool autoGenerateBindingRedirects;
+            bool.TryParse(projectMgr.BuildProject.GetPropertyValue("AutoGenerateBindingRedirects"), out autoGenerateBindingRedirects);
+
 #if FX_ATLEAST_45
-            // update FSharp.Core only if we are addressing '.NETFramework' as opposed to e.g. Silverlight or Portable
-            if (frameworkName.Identifier == ".NETFramework")
```

Comment:
This is called when the TargetFramework property is modified in the UI.  The UI only allows transitions specified in the SupportedRuntimes.Xml file.  So it seems to me that this won't be when the targetruntime is portable.  The purpose is to set the FSharpCore to the appropriate default fsharp.core for the target framework.  I could easily be wrong the code is kind of a rats nest.  I can add back the old check just in case, but I believe it is redundant.